### PR TITLE
Added task 1

### PR DESCRIPTION
**Pull request by DmitryZusko**
Added first task

### CODE REVIEW

File: Task_1/ComplexNumber.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+namespace Task_1
+{
+    class ComplexNumber
+    {
+        public double real { get; set; }
+        public double imaginary { get; set; }
+        private const int PRECISION = 3;
+
+        //default constructor
+        public ComplexNumber()
```
[the remark this answers is not in the document]
Comment (DmitryZusko) [reply Task_1/ComplexNumber.cs:15]: But when constructor with arguments initialized, default constructor doesn't work without writing it in code.

File: Task_1/ComplexNumber.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+        public ComplexNumber(double real, double imaginary)
+        {
+            this.real = real;
+            this.imaginary = imaginary;
+        }
+
+        public ComplexNumber(string complex)
+        {
+            //Should I use operation .ToList(), if Split returns array?
```
[the remark this answers is not in the document]
Comment (DmitryZusko) [reply Task_1/ComplexNumber.cs:41]: Ok

**Comment by DmitryZusko** [comment]
In my last commit were next changes:
overwrote operators (suck as +, -, *);
overwrote method ToString();
implemented interfaces IComparable<>, IEquatable<>

So now I can use ComplexNumber class with custom Generic classes

File: Task_1/ComplexNumber.cs [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+
+        public static ComplexNumber operator +(ComplexNumber complex1, ComplexNumber complex2)
+        {
+            return new ComplexNumber { Real = complex1.Real + complex2.Real, Imaginary = complex1.Imaginary + complex2.Imaginary };
+        }
+
+        public static ComplexNumber operator -(ComplexNumber complex1, ComplexNumber complex2)
+        {
+            var complex3 = new ComplexNumber(complex1.Real - complex2.Real, complex1.Imaginary - complex2.Imaginary);
```
Comment (mornado) [inline Task_1/ComplexNumber.cs:117]: Useless line.)

File: Task_1/ComplexNumber.cs [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            return new ComplexNumber { Real = complex1.Real - complex2.Real, Imaginary = complex1.Imaginary - complex2.Imaginary };
+        }
+
+        public static ComplexNumber operator *(ComplexNumber complex1, ComplexNumber complex2)
+        {
+            var complex3 = new ComplexNumber();
+            complex3.Real = Math.Round(complex1.Real * complex2.Real - complex1.Imaginary * complex2.Imaginary, PRECISION);
```
Comment (mornado) [inline Task_1/ComplexNumber.cs:124]: Can be processed during initialisation of complex3 variable.

**Review by mornado — CHANGES_REQUESTED** [review]

**Review by mornado — APPROVED** [review]

**Pull request merged**